src/components/mahjongBoard/MahjongBoard.js

```diff
@@ -1,66 +1,134 @@
-import React, { useState, useEffect }  from "react";
+import React, { useState, useEffect, useRef }  from "react";
 import MahjongItem from "../mahjongItem/MahjongItem";
+import PrimeNumbersArray from "../../helpers/utils";
 import style from "./mahjongBoard.scss";
 import classNames from "classnames";
 
-const PrimeNumbersArray = () => {
-	const primeNumbersArray = [];
-
-	// get array of prime numbers in range 160
-	for (let i = 1; i <= 160; i++) {
-		let flag = 0;
-		for (let j = 2; j < i; j++) {
-			if (0 === i % j) {
-				flag = 1;
-				break;
-			}
-		}
-		if (i > 1 && 0 === flag) {
-			primeNumbersArray.push(i);
-		}
-	}
-
-	// create random modifier to slice array of length 16
-	let randomModifier = Math.floor(Math.random() * 10);
-
-	// slice array to have length 16 with random modifier, duplicate values to have length 32 and sort random
-	return primeNumbersArray.slice(randomModifier, +randomModifier + 16).flatMap( i => [i,i]);
-};
-
 const MahjongBoard = () => {
 
-	const numbersArray = PrimeNumbersArray();
-	numbersArray.sort(function(a, b){return 0.5 - Math.random()});
-
 	const classes = classNames(
 		style.link,
 		'mahjong-board'
 	);
 
-	const initialState = [];
-	for (let i = 0; i < numbersArray.length; i++) {
-		initialState.push({
+	const cardsNumber = 32;
+	const initialState0 = [];
+	for (let i = 0; i < cardsNumber; i++) {
+		initialState0.push({
 			id: i,
-			key: numbersArray[i],
+			key: i,
 			status: 'visible',
 			tempStatus: '',
 		})
 	}
-	const stateHidden = initialState.map(e => ({...e, status: 'hidden'}));
 
-	const [boardState, setBoardState] = useState(initialState);
+	const [boardState, setBoardState] = useState(initialState0);
+
+	const clickCount = useRef(0);
+	const recentClickNumber = useRef(null);
+	const recentClickId = useRef(null);
 
 	useEffect(() => {
-		localStorage.setItem('clickCount', 0);
-		localStorage.setItem('recentClickNumber', null);
-		localStorage.setItem('recentClickId', null);
+		recentClickNumber.current = null;
+		recentClickId.current = null;
+		clickCount.current = 0;
+		const numbersArray = PrimeNumbersArray(160);
```

Comment:
Number range should be limited to (1,55) - from 1 (not included) to 55 (not included)